pmxbot/slack.py

```diff
@@ -64,14 +65,19 @@ def get_email_username_map(self, ttl_hash=get_ttl_hash()):
             }
 
     def start(self):
-        res = self.slack.rtm_connect()
-        assert res, "Error connecting"
         self.init_schedule(self.scheduler)
+        threading.Thread(target=self.run_scheduler_loop).start()
+
+        @self.slack.on("message")
+        def handle_payload(client, event):
+            self.handle_message(event)
```

Comment:
Did you consider instead using `self.slack.on("message")(self.handle_message)` and updating `handle_message` to accept `(client, event)`?

Reply:
agreed, actioned

Reply:
hmm, apparently not possible as slack validates the function signature:
```
    raise SlackClientError(error)
slack_sdk.errors.SlackClientError: The listener 'handle_message' must accept two args: client, event (actual: self, client, msg)
```

We can't really make `handle_message` static since it uses `self` all over